[BUG] favicon.ico is not resolved relative to PAPERLESS_STATICDIR

### Description

Currently, Paperless [loads](https://github.com/paperless-ngx/paperless-ngx/blob/dev/src/paperless/views.py#L97) `favicon.ico` relative to the location of the `views.py` file (`f{os.path.dirname(__file__)}/static/paperless/img/favicon.ico"`). This can fail when the static files have been relocated to a location separate from the sources.

Happy to contribute a fix, but checking for the desired approach first. Should Paperless
1) only look relative to `PAPERLESS_STATICDIR` / `STATIC_ROOT`, or
2) first look for the file in the current location and only if that fails look at the location above?

### Steps to reproduce

1. Set `PAPERLESS_STATICDIR` to a custom location
2. Run `collectstatic`
3. Start paperless
4. Observe an error like `FileNotFoundError: [Errno 2] No such file or directory: '/opt/homebrew/Cellar/paperless-ngx/2.16.2_1/libexec/lib/python3.13/site-packages/paperless/static/paperless/img/favicon.ico'` in the log

### Webserver logs

```bash
[2025-06-02 20:57:20,531] [ERROR] [django.request] Internal Server Error: /favicon.ico
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/paperless-ngx/2.16.2_1/libexec/lib/python3.13/site-packages/asgiref/sync.py", line 518, in thread_handler
    raise exc_info[1]
  File "/opt/homebrew/Cellar/paperless-ngx/2.16.2_1/libexec/lib/python3.13/site-packages/django/core/handlers/exception.py", line 42, in inner
    response = await get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/paperless-ngx/2.16.2_1/libexec/lib/python3.13/site-packages/asgiref/sync.py", line 518, in thread_handler
    raise exc_info[1]
  File "/opt/homebrew/Cellar/paperless-ngx/2.16.2_1/libexec/lib/python3.13/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
    response = await wrapped_callback(
               ^^^^^^^^^^^^^^^^^^^^^^^
        request, *callback_args, **callback_kwargs
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/homebrew/Cellar/paperless-ngx/2.16.2_1/libexec/lib/python3.13/site-packages/asgiref/sync.py", line 468, in __call__
    ret = await asyncio.shield(exec_coro)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/paperless-ngx/2.16.2_1/libexec/lib/python3.13/site-packages/asgiref/current_thread_executor.py", line 40, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/opt/homebrew/Cellar/paperless-ngx/2.16.2_1/libexec/lib/python3.13/site-packages/asgiref/sync.py", line 522, in thread_handler
    return func(*args, **kwargs)
  File "/opt/homebrew/Cellar/paperless-ngx/2.16.2_1/libexec/lib/python3.13/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/paperless-ngx/2.16.2_1/libexec/lib/python3.13/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
  File "/opt/homebrew/Cellar/paperless-ngx/2.16.2_1/libexec/lib/python3.13/site-packages/paperless/views.py", line 103, in get
    with open(favicon, "rb") as f:
         ~~~~^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/opt/homebrew/Cellar/paperless-ngx/2.16.2_1/libexec/lib/python3.13/site-packages/paperless/static/paperless/img/favicon.ico'
```

### Browser logs

```bash

```

### Paperless-ngx version

2.16.2

### Host OS

macOS

### Installation method

Bare metal

### System status

```json

```

### Browser

_No response_

### Configuration changes

_No response_

### Please confirm the following

- [x] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [x] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [x] I have already searched for relevant existing issues and discussions before opening this report.
- [x] I have updated the title field above with a concise description.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -239,9 +239,6 @@ lint.per-file-ignores."src/paperless/checks.py" = [
 lint.per-file-ignores."src/paperless/settings.py" = [
   "PTH",
 ] # TODO Enable & remove
-lint.per-file-ignores."src/paperless/views.py" = [
-  "PTH",
-] # TODO Enable & remove
 lint.per-file-ignores."src/paperless_mail/mail.py" = [
   "PTH",
 ] # TODO Enable & remove

--- a/src/paperless/tests/test_views.py
+++ b/src/paperless/tests/test_views.py
@@ -1,11 +1,9 @@
 import tempfile
 from pathlib import Path
 
-import pytest
 from django.conf import settings
 
 
-@pytest.mark.django_db
 def test_favicon_view(client):
     with tempfile.TemporaryDirectory() as tmpdir:
         static_dir = Path(tmpdir)

--- a/src/paperless/tests/test_views.py
+++ b/src/paperless/tests/test_views.py
@@ -1,0 +1,27 @@
+import tempfile
+from pathlib import Path
+
+import pytest
+from django.conf import settings
+
+
+@pytest.mark.django_db
+def test_favicon_view(client):
+    with tempfile.TemporaryDirectory() as tmpdir:
+        static_dir = Path(tmpdir)
+        favicon_path = static_dir / "paperless" / "img" / "favicon.ico"
+        favicon_path.parent.mkdir(parents=True, exist_ok=True)
+        favicon_path.write_bytes(b"FAKE ICON DATA")
+
+        settings.STATIC_ROOT = static_dir
+
+        response = client.get("/favicon.ico")
+        assert response.status_code == 200
+        assert response["Content-Type"] == "image/x-icon"
+        assert b"".join(response.streaming_content) == b"FAKE ICON DATA"
+
+
+def test_favicon_view_missing_file(client):
+    settings.STATIC_ROOT = Path(tempfile.mkdtemp())
+    response = client.get("/favicon.ico")
+    assert response.status_code == 404

--- a/src/paperless/views.py
+++ b/src/paperless/views.py
@@ -1,5 +1,5 @@
-import os
 from collections import OrderedDict
+from pathlib import Path
 
 from allauth.mfa import signals
 from allauth.mfa.adapter import get_adapter as get_mfa_adapter
@@ -11,8 +11,9 @@ from allauth.socialaccount.adapter import get_adapter
 from allauth.socialaccount.models import SocialAccount
 from django.contrib.auth.models import Group
 from django.contrib.auth.models import User
+from django.contrib.staticfiles.storage import staticfiles_storage
 from django.db.models.functions import Lower
-from django.http import HttpResponse
+from django.http import FileResponse
 from django.http import HttpResponseBadRequest
 from django.http import HttpResponseForbidden
 from django.http import HttpResponseNotFound
@@ -92,16 +93,12 @@ class StandardPagination(PageNumberPagination):
 
 
 class FaviconView(View):
-    def get(self, request, *args, **kwargs):  # pragma: no cover
-        favicon = os.path.join(
-            os.path.dirname(__file__),
-            "static",
-            "paperless",
-            "img",
-            "favicon.ico",
-        )
-        with open(favicon, "rb") as f:
-            return HttpResponse(f, content_type="image/x-icon")
+    def get(self, request, *args, **kwargs):
+        try:
+            path = Path(staticfiles_storage.path("paperless/img/favicon.ico"))
+            return FileResponse(path.open("rb"), content_type="image/x-icon")
+        except FileNotFoundError:
+            return HttpResponseNotFound("favicon.ico not found")
 
 
 class UserViewSet(ModelViewSet):

